guides/micronaut-oracle-cloud-streaming/chess-listener/groovy/src/main/groovy/example/micronaut/chess/GameService.groovy

```diff
@@ -0,0 +1,90 @@
+package example.micronaut.chess
+
+import example.micronaut.chess.dto.GameDTO
+import example.micronaut.chess.dto.GameStateDTO
+import example.micronaut.chess.entity.Game
+import example.micronaut.chess.entity.GameState
+import example.micronaut.chess.repository.GameRepository
+import example.micronaut.chess.repository.GameStateRepository
+import groovy.transform.CompileStatic
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import javax.inject.Singleton
+import javax.transaction.Transactional
+
+/**
+ * Service to encapsulate business logic.
+ */
+@Singleton
+@Transactional
```

Comment:
Why the `@Transactional` annotation here? It should be just in the repositories? We are not saving two both repositories in a single transaction.

Reply:
This is habit mostly - even if not doing multiple changes it's best to start using transactions (unless there's a performance impact/concern) and then later when things get more complex there's no need to add transactionality.

I don't think it's appropriate to add `@Transactional` to repositories, since different use cases can require different transaction rules. I prefer a pattern (transactional facade) where services use repositories and define the tx logic.